inst/stan/functions/pmfs.stan

```diff
@@ -54,10 +53,24 @@ vector combine_pmfs(vector fixed_pmf, real[] pmf_mu, real[] pmf_sigma, int[] pmf
   } else if (n_variable > 0) {
     pmf[1] = 1;
   }
-  for (s in 1:n_variable) {
-    vector[pmf_n[s]] variable_pmf;
-    variable_pmf = discretised_pmf(pmf_mu[s], pmf_sigma[s], pmf_n[s], dist[s], left_truncate);
-    pmf = convolve(pmf, variable_pmf, len);
+  if (n_variable > 0) {
+    for (s in 1:n_variable) {
+      vector[pmf_n[s]] variable_pmf;
+      variable_pmf = discretised_pmf(pmf_mu[s], pmf_sigma[s], pmf_n[s], dist[s]);
+      pmf = convolve(pmf, variable_pmf, len);
+    }
+  }
+  if (left_truncate) {
+    real norm;
+    for (i in 1:left_truncate) {
+      pmf[i] = 0;
+    }
+    norm = sum(pmf);
+    if (left_truncate < len) {
+      for (i in left_truncate:len) {
```

Comment:
this seems like it could be causing a slow down can it now be vectorised using `rep_vector` and `{}` to construct a new combined vector?

Reply:
Something like this? https://github.com/epiforecasts/EpiNow2/pull/363/commits/d4e89c2674e3c929c7554c74c249292164cca30d

Reply:
yes